lib/remap-lang/src/expression/arithmetic.rs

```diff
@@ -18,6 +18,13 @@ impl Expression for Arithmetic {
     fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
         use Operator::*;
 
+        if matches!(self.op, ErrorOr) {
+            return self
+                .lhs
+                .execute(state, object)
+                .or_else(|_| self.rhs.execute(state, object));
+        }
+
         let lhs = self.lhs.execute(state, object)?;
         let rhs = self.rhs.execute(state, object)?;
```

Comment:
I realise this isn't part of this PR, but isn't this wrong for `Or` and `And`? We want to short circuit execution of `rhs` depending on the result of `lhs`.

Reply:
If so, it might be better to restructure this a bit such that we execute `lhs` first, then have a match for `ErrorOr`, `Or` and `And` that handles the rhs accordingly and returns. Otherwise it executes `rhs` and proceeds as normal with the other operators..

Reply:
Yep, we need to fix that. Mind opening an issue to track this before we ship 0.12 @FungusHumungus?

Reply:
Apparently I already filed an issue for this a while ago 🙈, tracking in https://github.com/timberio/vector/issues/4706.